runtimes/frameless-runtime/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+	//		 pub babe: Babe,
+	//	 }
+	// }
+}
+
+/// This runtime version.
+pub const VERSION: RuntimeVersion = RuntimeVersion {
+	spec_name: create_runtime_str!("frameless-runtime"),
+	impl_name: create_runtime_str!("frameless-runtime"),
+	authoring_version: 1,
+	spec_version: 1,
+	impl_version: 1,
+	apis: RUNTIME_API_VERSIONS,
+	transaction_version: 1,
+};
+
+/// The version infromation used to identify this runtime when compiled natively.
+#[cfg(feature = "std")]
+pub fn native_version() -> NativeVersion {
+	NativeVersion {
+		runtime_version: VERSION,
+		can_author_with: Default::default(),
+	}
+}
+
+/// The main struct in this module. In frame this comes from `construct_runtime!`
+pub struct Runtime;
+
+impl GetNodeBlockType for Runtime {
+	type NodeBlock = opaque::Block;
+}
+
+impl GetRuntimeBlockType for Runtime {
+	type RuntimeBlock = opaque::Block;
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Default))]
+pub struct GenesisConfig;
+
+#[cfg(feature = "std")]
+impl BuildStorage for GenesisConfig {
+	fn assimilate_storage(&self, storage: &mut Storage) -> Result<(), String> {
+		// Declare the storage items we need
+		let storage_items = vec![
+			(ONLY_KEY.encode(), false.encode()),
+			(well_known_keys::CODE.into(), WASM_BINARY.to_vec()),
+		];
+
+		// Put them into genesis storage
+		storage.top.extend(
+			storage_items.into_iter()
+		);
+
+		Ok(())
+	}
+}
+
+/// The address format for describing accounts.
+// pub type Address = <Indices as StaticLookup>::Source;
+/// Block header type as expected by this runtime.
+pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
+/// Block type as expected by this runtime.
+pub type Block = generic::Block<Header, FramelessTransaction>;
+/// A Block signed with a Justification
+pub type SignedBlock = generic::SignedBlock<Block>;
+/// BlockId type as expected by this runtime.
+pub type BlockId = generic::BlockId<Block>;
+
+pub const ONLY_KEY: [u8; 1] = [0];
+pub const HEADER_KEY: [u8; 6] = *b"header";
+
+// I guess we won't need any of this when using our own unchecked extrinsic type
+// The SignedExtension to the basic transaction logic.
+// pub type SignedExtra = (
+//	 system::CheckVersion<Runtime>,
+//	 system::CheckGenesis<Runtime>,
+// );
+/// Unchecked extrinsic type as expected by this runtime.
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, parity_util_mem::MallocSizeOf))]
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub enum FramelessTransaction {
+	Set,
+	Clear,
+	Toggle,
+	//TODO in future define call
+}
+
+impl Extrinsic for FramelessTransaction {
+	type Call = ();
+	type SignaturePayload = ();
+
+	fn new(_call: Self::Call, _signed_data: Option<Self::SignaturePayload>) -> Option<Self> {
+		Some(Self::Toggle)
+	}
+}
+
+impl_runtime_apis! {
+	// https://substrate.dev/rustdocs/master/sp_api/trait.Core.html
+	impl sp_api::Core<Block> for Runtime {
+		fn version() -> RuntimeVersion {
+			VERSION
+		}
+
+		fn execute_block(block: Block) {
```

Comment:
This function is wrong. You don't call `intialize_block` and `finalize_block`.

Reply:
I've added a commit which calls those functions, but the behavior is the same as before. Maybe I called them incorrectly?

Reply:
Your `apply_extrinsic` function is not doing anything. And your `execute_block` function handles the transactions. I'm not sure how you expect this to work? You need to do the exactly same operations while building and importing the block.